storage/watch.go

```diff
@@ -0,0 +1,54 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// Watcher uses to watch on key-value store.
+type Watcher interface {
```

Comment:
WatchableKV is the platform who can cancel a existing watcher on its key-value store. If Watcher is the one, it needs to use some function in WatchableKV to cancel itself anyway, so I think it is indirect.